software/controller/controller.sh

```diff
@@ -0,0 +1,168 @@
+cd "$(dirname "$0")"
+
+# Check if Linux
+PLATFORM="$(uname -s)"
+if [ $PLATFORM != "Linux" ]; then
+  echo "Error: This script only supports 'Linux'. You have $PLATFORM."
+  exit 1
+fi
+
+#########
+# UTILS #
+#########
+
+print_help() {
+    cat <<EOF
+RespiraWorks controller build utilities. Will build and run most tests. Will also build embedded controller code.
+The following options are provided:
+  --help        Display this dialog
+  --install     Install platformio and configure udev rules for deployment
+  --clean       Clean build directories
+  --test        Builds and runs all tests locally
+    [--no-checks]     - do not run static checks (yes, it's to annoy you!)
+  --run         Builds and deploys firmware to controller (there can be only one, use deploy script manually for others)
+EOF
+}
+
+clean_dir() {
+  dir_name=$1
+  if [ -d "$dir_name" ]; then
+    echo "Removing $dir_name"
+    rm -rf "$dir_name"
+    return 0
+  elif [ -f "$dir_name" ]; then
+    echo "File with this name already exists, not a directory."
+    return 1
+  fi
+}
+
+########
+# HELP #
+########
+
+if [ "$1" == "--help" ]; then
+  print_help
+  exit 0
+fi
+
+###########
+# INSTALL #
+###########
+
+if [ "$1" == "--install" ]; then
+  if [ "$EUID" -eq 0 ] && [ "$2" != "-f" ]; then
+    echo "Please do not run install with root privileges!"
+    exit 1
+  fi
+
+  pushd "$HOME"
+  python3 -c "$(curl -fsSL https://raw.githubusercontent.com/platformio/platformio/master/scripts/get-platformio.py)"
+  echo 'export PATH=$PATH:~/.platformio/penv/bin' >> ~/.profile
+  popd
+
+  curl -fsSL https://raw.githubusercontent.com/platformio/platformio-core/master/scripts/99-platformio-udev.rules | sudo tee /etc/udev/rules.d/99-platformio-udev.rules
+  sudo service udev restart
+  sudo usermod -a -G dialout "$USER"
+  sudo usermod -a -G plugdev "$USER"
+
+  echo "Updated udev rules. You might have to restart your machine for changes to become effective."
+
+  exit 0
+fi
+
+#########
+# CLEAN #
+#########
+
+if [ "$1" == "--clean" ]; then
+  clean_dir .pio
+  exit 0
+fi
+
+########
+# TEST #
+########
+
+if [ "$1" == "--test" ]; then
+  # Controller unit tests on native.
+  pio test -e native
+
+  # Make sure controller integration tests build for target platform.
+  # TODO(martukas) This will actually have to be build and tested on Jenkins and not here
+  INTEGRATION_TEST_H=idle_test.h pio run -e integration-test
+  INTEGRATION_TEST_H=buzzer_test.h TEST_PARAM_1=0.0f TEST_PARAM_2=1.0f pio run -e integration-test
+  INTEGRATION_TEST_H=blower_test.h TEST_PARAM_1=0.0f TEST_PARAM_2=1.0f pio run -e integration-test
+  INTEGRATION_TEST_H=stepper_test.h TEST_PARAM_1=0 TEST_PARAM_2=90.0f pio run -e integration-test
+  INTEGRATION_TEST_H=pinch_valve_test.h TEST_PARAM_1=0 pio run -e integration-test
+  INTEGRATION_TEST_H=psol_test.h pio run -e integration-test
+
+  # Make sure controller builds for target platform.
+  pio run -e stm32
```

Comment:
We will need to update this in case we change the default environment in platformio.ini, is this what we want, or should this rather build the default environment?

Reply:
I believe this was here all along, before I renamed the script. How else could we do it?